pkg/kritis/review/review.go

```diff
@@ -61,8 +61,35 @@ func generateGapAllowlist(gaps []v1beta1.GenericAttestationPolicy) []string {
 	return allowlist
 }
 
+// ReviewImageWithGAP reviews single image against a single generic attestation policy
+func (r Reviewer) ReviewImageWithGAP(image string, gap v1beta1.GenericAttestationPolicy, c metadata.ReadOnlyClient) (bool, []string, error) {
+	glog.Infof("Validating against GenericAttestationPolicy %s", gap.Name)
+
+	// Get all AttestationAuthorities in this policy.
```

Comment:
This whole section should be in ReviewGAP.  Then you can error out immediately (before even attempting to review any images) if there are no auths.  (Also no need to call getAttestationAuthoritiesForGAP() for every image.)

Reply:
Per discussion, adding `checkGAPs()` in the beginning of `ReviewGAP()` to throw out error early. Minor: now it's still calling `getAttestationAuthoritiesForGAP()` for every image.